watchtower-plugin/src/net/http.rs

```diff
@@ -154,6 +154,74 @@ pub async fn send_appointment(
     }
 }
 
+/// Handles a ping request to the tower.
+pub async fn ping(tower_net_addr: &NetAddr) -> Result<(), RequestError> {
+    let response = get_request(tower_net_addr, Endpoint::Ping, &None).await?;
```

Comment:
You need to honor the proxy, not just assume it will be `None`. This means the proxy info needs to be passed to the `ping` function and forwarded to the resquest

Reply:
Oh okay! Got it.